mmedit/models/editors/nafnet/baseline_net.py

```diff
@@ -0,0 +1,287 @@
+            in_channels=img_channel,
+            out_channels=mid_channels,
+            kernel_size=3,
+            padding=1,
+            stride=1,
+            groups=1,
+            bias=True)
+        self.ending = nn.Conv2d(
+            in_channels=mid_channels,
+            out_channels=img_channel,
+            kernel_size=3,
+            padding=1,
+            stride=1,
+            groups=1,
+            bias=True)
+
+        self.encoders = nn.ModuleList()
+        self.decoders = nn.ModuleList()
+        self.middle_blks = nn.ModuleList()
+        self.ups = nn.ModuleList()
+        self.downs = nn.ModuleList()
+
+        chan = mid_channels
+        for num in enc_blk_nums:
+            self.encoders.append(
+                nn.Sequential(*[
+                    BaselineBlock(chan, dw_expand, ffn_expand)
+                    for _ in range(num)
+                ]))
+            self.downs.append(nn.Conv2d(chan, 2 * chan, 2, 2))
+            chan = chan * 2
+
+        self.middle_blks = \
+            nn.Sequential(
+                *[BaselineBlock(chan, dw_expand, ffn_expand)
+                    for _ in range(middle_blk_num)]
+            )
+
+        for num in dec_blk_nums:
+            self.ups.append(
+                nn.Sequential(
+                    nn.Conv2d(chan, chan * 2, 1, bias=False),
+                    nn.PixelShuffle(2)))
+            chan = chan // 2
+            self.decoders.append(
+                nn.Sequential(*[
+                    BaselineBlock(chan, dw_expand, ffn_expand)
+                    for _ in range(num)
+                ]))
+
+        self.padder_size = 2**len(self.encoders)
+
+    def forward(self, inp):
+        """Forward function.
+
+        args:
+            inp: input tensor image with (B, C, H, W) shape
+        """
+        B, C, H, W = inp.shape
+        inp = self.check_image_size(inp)
+
+        x = self.intro(inp)
+
+        encs = []
+
+        for encoder, down in zip(self.encoders, self.downs):
+            x = encoder(x)
+            encs.append(x)
+            x = down(x)
+
+        x = self.middle_blks(x)
+
+        for decoder, up, enc_skip in zip(self.decoders, self.ups, encs[::-1]):
+            x = up(x)
+            x = x + enc_skip
+            x = decoder(x)
+
+        x = self.ending(x)
+        x = x + inp
+
+        return x[:, :, :H, :W]
+
+    def check_image_size(self, x):
+        """Check image size and pad images so that it has enough dimension do
+        downsample.
+
+        args:
+            x: input tensor image with (B, C, H, W) shape.
+        """
+        _, _, h, w = x.size()
+        mod_pad_h = (self.padder_size -
+                     h % self.padder_size) % self.padder_size
+        mod_pad_w = (self.padder_size -
+                     w % self.padder_size) % self.padder_size
+        x = F.pad(x, (0, mod_pad_w, 0, mod_pad_h))
+        return x
+
+
+@MODELS.register_module()
+class BaselineLocal(Local_Base, Baseline):
```

Comment:
The class name `BaselineLocal` is not suitable.

Reply:
Rename it to NAFBaselineLocal